primitives/src/channel.rs

```diff
@@ -74,6 +75,18 @@ impl fmt::Display for ChannelId {
     }
 }
 
+impl FromStr for ChannelId {
```

Comment:
Could you unify the `channel_id_from_str` and this impl of `FromStr`?
The `channel_id_from_str` doesn't check for `0x` prefix which means they act differently.

Idk what will look ok given the serialization is handled differently, however a lot of the code could be removed if we do it.